File: SistemaEventosAngular/src/app/interfaces/reservacion.ts
import { Evento } from './evento';
import { Cliente } from './cliente';

export interface Reservacion {
  reservacionId: number;
  eventoId: number;
  evento?: Evento;
  clienteId: number;
  cliente?: Cliente;
  fechaReservacion: Date;
  estado: string;
}

File: SistemaEventos/Models/Reservacion.cs
namespace SistemaEventos.Models
{
    using System.ComponentModel.DataAnnotations;

    public class Reservacion
    {
        public int ReservacionId { get; set; }
        public int EventoId { get; set; }
        public Evento Evento { get; set; }

        public int ClienteId { get; set; }
        public Cliente Cliente { get; set; }

        public DateTime FechaReservacion { get; set; }
        public string Estado { get; set; } 
    }
}
